src/withStatechart.js

```diff
@@ -11,13 +11,13 @@ const withStatechart = (statechart, options = {}) => Component => {
     constructor(props) {
       super(props)
 
-      const initialMachineState = this.props.initialMachineState
-        ? State.from(this.props.initialMachineState)
+      const machineState = this.props.initialMachineState
```

Comment:
nit: `const machineState = this.props.initialMachineState || this.machine.initialState`